### PR TITLE
Fix failing tests with WebGL Validation

### LINKED ISSUE #2373: Test failures with WebGL validation
Not going to hold up the 1.5 release for these. @bagnell can you look at these before the 1.6 release?

```
Scene/RectanglePrimitive renders bounding volume with debugShowBoundingVolume. (show summary)
RuntimeError: WebGL Error: INVALID_OPERATION: drawElements(4, 24, 5123, 0);
Error
    at new RuntimeError (http://localhost:8080/Source/Core/RuntimeError.js:42:19)
    at throwOnError (http://localhost:8080/Source/Renderer/Context.js:123:19)
    at Object.drawElements (http://localhost:8080/Source/Renderer/Context.js:149:17)
    at continueDraw (http://localhost:8080/Source/Renderer/Context.js:1834:25)
    at Context.draw (http://localhost:8080/Source/Renderer/Context.js:1861:9)
    at DrawCommand.execute (http://localhost:8080/Source/Renderer/DrawCommand.js:185:17)
    at executeCommand (http://localhost:8080/Source/Scene/Scene.js:1050:21)
    at executeCommands (http://localhost:8080/Source/Scene/Scene.js:1279:17)
    at render (http://localhost:8080/Source/Scene/Scene.js:1377:9)
    at Scene.render (http://localhost:8080/Source/Scene/Scene.js:1413:13)
Scene/Sun draws in 3D. (show summary)
DeveloperError: Framebuffer is not complete. Incomplete attachment: at least one attachment point with a renderbuffer or texture attached has its attached object no longer in existence or has an attached image with a width or height of zero, or the color attachment point has a non-color-renderable image attached, or the depth attachment point has a non-depth-renderable image attached, or the stencil attachment point has a non-stencil-renderable image attached. Color-renderable formats include GL_RGBA4, GL_RGB5_A1, and GL_RGB565. GL_DEPTH_COMPONENT16 is the only depth-renderable format. GL_STENCIL_INDEX8 is the only stencil-renderable format.
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:43:19)
    at validateFramebuffer (http://localhost:8080/Source/Renderer/Context.js:1690:23)
    at bindFramebuffer (http://localhost:8080/Source/Renderer/Context.js:1717:17)
    at Context.clear (http://localhost:8080/Source/Renderer/Context.js:1774:9)
    at ClearCommand.execute (http://localhost:8080/Source/Renderer/ClearCommand.js:98:17)
    at Sun.update (http://localhost:8080/Source/Scene/Sun.js:203:26)
    at executeCommands (http://localhost:8080/Source/Scene/Scene.js:1182:87)
    at render (http://localhost:8080/Source/Scene/Scene.js:1377:9)
    at Scene.render (http://localhost:8080/Source/Scene/Scene.js:1413:13)
    at scene.renderForSpecs (http://localhost:8080/Specs/createScene.js:42:19)
Scene/Sun draws in Columbus view. (show summary)
DeveloperError: Framebuffer is not complete. Incomplete attachment: at least one attachment point with a renderbuffer or texture attached has its attached object no longer in existence or has an attached image with a width or height of zero, or the color attachment point has a non-color-renderable image attached, or the depth attachment point has a non-depth-renderable image attached, or the stencil attachment point has a non-stencil-renderable image attached. Color-renderable formats include GL_RGBA4, GL_RGB5_A1, and GL_RGB565. GL_DEPTH_COMPONENT16 is the only depth-renderable format. GL_STENCIL_INDEX8 is the only stencil-renderable format.
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:43:19)
    at validateFramebuffer (http://localhost:8080/Source/Renderer/Context.js:1690:23)
    at bindFramebuffer (http://localhost:8080/Source/Renderer/Context.js:1717:17)
    at Context.clear (http://localhost:8080/Source/Renderer/Context.js:1774:9)
    at ClearCommand.execute (http://localhost:8080/Source/Renderer/ClearCommand.js:98:17)
    at Sun.update (http://localhost:8080/Source/Scene/Sun.js:203:26)
    at executeCommands (http://localhost:8080/Source/Scene/Scene.js:1182:87)
    at render (http://localhost:8080/Source/Scene/Scene.js:1377:9)
    at Scene.render (http://localhost:8080/Source/Scene/Scene.js:1413:13)
    at scene.renderForSpecs (http://localhost:8080/Specs/createScene.js:42:19)
Scene/Sun draws without lens flare. (show summary)
DeveloperError: Framebuffer is not complete. Incomplete attachment: at least one attachment point with a renderbuffer or texture attached has its attached object no longer in existence or has an attached image with a width or height of zero, or the color attachment point has a non-color-renderable image attached, or the depth attachment point has a non-depth-renderable image attached, or the stencil attachment point has a non-stencil-renderable image attached. Color-renderable formats include GL_RGBA4, GL_RGB5_A1, and GL_RGB565. GL_DEPTH_COMPONENT16 is the only depth-renderable format. GL_STENCIL_INDEX8 is the only stencil-renderable format.
Error
    at new DeveloperError (http://localhost:8080/Source/Core/DeveloperError.js:43:19)
    at validateFramebuffer (http://localhost:8080/Source/Renderer/Context.js:1690:23)
    at bindFramebuffer (http://localhost:8080/Source/Renderer/Context.js:1717:17)
    at Context.clear (http://localhost:8080/Source/Renderer/Context.js:1774:9)
    at ClearCommand.execute (http://localhost:8080/Source/Renderer/ClearCommand.js:98:17)
    at Sun.update (http://localhost:8080/Source/Scene/Sun.js:203:26)
    at executeCommands (http://localhost:8080/Source/Scene/Scene.js:1182:87)
    at render (http://localhost:8080/Source/Scene/Scene.js:1377:9)
    at Scene.render (http://localhost:8080/Source/Scene/Scene.js:1413:13)
    at scene.renderForSpecs (http://localhost:8080/Specs/createScene.js:42:19)
```

### PR DESCRIPTION
**Pull request by bagnell**
Fixes #2373.

### CODE REVIEW

File: Specs/createScene.js [hunk 1]
```diff
@@ -16,9 +16,14 @@ define([
     "use strict";
 
     function createScene(options) {
+        options = defaultValue(options, {});
+
+        var canvas = defaultValue(options.canvas, createCanvas());
+        options.canvas = undefined;
+
         options = clone(defaultValue(options, {}), true);
```
Comment (pjcozzi) [inline Specs/createScene.js]: Can't this just be `options = clone(options, true);` now?

File: Specs/createScene.js [hunk 2]
```diff
@@ -16,9 +16,14 @@ define([
     "use strict";
 
     function createScene(options) {
+        options = defaultValue(options, {});
+
+        var canvas = defaultValue(options.canvas, createCanvas());
+        options.canvas = undefined;
```
Comment (pjcozzi) [inline Specs/createScene.js]: Can we add a comment about what is going on here, e.g., we don't want to clone the HTML canvas element.

**Comment by pjcozzi** [comment]
Everything looks good otherwise.

File: Specs/createScene.js [hunk 3]
```diff
@@ -16,9 +16,14 @@ define([
     "use strict";
 
     function createScene(options) {
+        options = defaultValue(options, {});
+
+        var canvas = defaultValue(options.canvas, createCanvas());
```
Comment (mramato) [inline Specs/createScene.js]: This should be `defined(options.canvas) ? options.canvas : createCanvas()`. The way it's currently written, if `options.canvas` is specified, we add an extra canvas to the document and it never gets removed (which could technically interfere with future specs).

**Comment by bagnell** [comment]
@pjcozzi @mramato This is ready for another review.

**Comment by pjcozzi** [comment]
Looks good.

**Pull request merged**